File: src-typescript/ch06/CeilingFan.ts
export class CeilingFan {

    static readonly HIGH: number = 3;
    static readonly MEDIUM: number = 2;
    static readonly LOW: number = 1;
    static readonly OFF: number = 0;

    place: string;
    speed: number;

    constructor(place: string) {
        this.place = place;
        this.speed = CeilingFan.OFF;
    }

    high(): void {
        this.speed = CeilingFan.HIGH;
        console.log(this.place + "의 선풍기 속도가 HIGH로 설정되었습니다.")
    }

    medium(): void {
        this.speed = CeilingFan.MEDIUM;
        console.log(this.place + "의 선풍기 속도가 MEDIUM으로 설정되었습니다.");
    }

    low(): void {
        this.speed = CeilingFan.LOW;
        console.log(this.place + "의 선풍기 속도가 LOW로 설정되었습니다.")
    }

    off(): void {
        this.speed = CeilingFan.OFF;
        console.log(this.place + "의 선풍기의 전원을 끕니다.")
    }

    getSpeed(): number {
        return this.speed;
    }
}

File: src/ch06/CeilingFan.java
package ch06;

public class CeilingFan {

    public static final int HIGH = 3;
    public static final int MEDIUM = 2;
    public static final int LOW = 1;
    public static final int OFF = 0;
    String place;
    int speed;

    public CeilingFan(String place) {
        this.place = place;
        speed = OFF;
    }

    public void high() {
        speed = HIGH;
        System.out.println(place + "의 선풍기 속도가 HIGH로 설정되었습니다.");
    }

    public void medium() {
        speed = MEDIUM;
        System.out.println(place + "의 선풍기 속도가 MEDIUM으로 설정되었습니다.");
    }

    public void low() {
        speed = LOW;
        System.out.println(place + "의 선풍기 속도가 LOW로 설정되었습니다.");
    }

    public void off() {
        speed = OFF;
        System.out.println(place + "의 선풍기의 전원을 끕니다.");
    }

    public int getSpeed() {
        return speed;
    }
}
